makedebian.pl

```diff
@@ -260,11 +260,17 @@
 open(SCRIPT, ">$postinstall_file");
 print SCRIPT <<EOF;
 #!/bin/sh
+justinstalled=1
+if [ -e "/etc/$baseproduct" ]; then
+	justinstalled=0
+fi
 inetd=`grep "^inetd=" /etc/$baseproduct/miniserv.conf 2>/dev/null | sed -e 's/inetd=//g'`
-if [ "\$1" = "upgrade" -a "\$1" != "abort-upgrade" ]; then
+if [ "\$1" = "configure" ]; then
 	# Upgrading the package, so stop the old Webmin properly
 	if [ "\$inetd" != "1" ]; then
-		/etc/$baseproduct/stop >/dev/null 2>&1 </dev/null
+		if [ -e "/etc/$baseproduct/.pre-install" ]; then
+			/etc/$baseproduct/.pre-install >/dev/null 2>&1 </dev/null
```

Comment:
N: In case of update or reinstall on all Webmin versions 1.996+ run `.pre-install` command, which will simply either run your older code from `/etc/webmin/.stop-init` script or in case of a `systemd` systems it will send _SIGSTOP_ on the main process (only!) while upgrading.